src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -75,7 +75,7 @@ export class Kernel implements IKernel {
     get kernelSocket(): Observable<KernelSocketInformation | undefined> {
         return this._kernelSocket.asObservable();
     }
-    private notebook?: INotebook;
+    public notebook?: INotebook;
```

Comment:
Exposing this property on the `IKernel` isn't ideal but allows us to reuse basically all the existing debugging code, since `IKernel` doesn't have the concept of `executeObservable`. I really didn't want to go down the path of forwarding more `INotebook` methods in `IKernel` for consumption in the interactive window. Do folks care either way?

Reply:
LGTM

Reply:
I think this was only so that you can use the `executeAsObservable`, but there are other issues there (side effects of running a cell that can update other cells).
hence we might have to just go with the standard IKernel.executeCell method.